OutOfSchool/OutOfSchool.WebApi/Controllers/V1/ApplicationController.cs

```diff
@@ -167,12 +168,64 @@ public async Task<IActionResult> GetByProviderId(Guid providerId, [FromQuery] Ap
 
         var applications = await applicationService.GetAllByProvider(providerId, filter).ConfigureAwait(false);
 
-        if (applications == null || !applications.Entities.Any())
+        return this.GetResultOfGetOperation(applications);
+    }
+
+    /// <summary>
+    /// Get collection of applications, that have a pending status.
+    /// </summary>
+    /// <param name="providerId">Provider id.</param>
+    /// <returns>List of applications.</returns>
+    /// <response code="200">Entities were found by given Id.</response>
+    /// <response code="204">No entity with given Id was found.</response>
+    /// <response code="400">Provider with given id was not found.</response>
+    /// <response code="401">User is unauthorized.</response>
+    /// <response code="500">If any server error occurs.</response>
+    [HasPermission(Permissions.ApplicationRead)]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(SearchResult<ApplicationDto>))]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    [HttpGet("/api/v{version:apiVersion}/provider/{providerId}/applications/pending")]
+    public async Task<IActionResult> GetPendingApplicationsByProviderId(Guid providerId)
+    {
+        // Find a standard provider by given id
```

Comment:
(Comments) Please avoid writing comments that duplicate code. I mean next example
`i += 1; // here we increase our counter by 1`
- it is really obvious and doesn't need to be explained in comment. 

There are several issues with all comments: 
- they increase code "reading" complexity (yes, you have to read more), 
- they become obsolete pretty fast (code is changed, comment is not - have seen such situaton many times in this project),
- they make refactoring more hard (or welcome obsolete comments),
- they are never checked by compiler or unit test and thus might be simply wrong (seen even in this PR).

Reply:
Fixed.